webrtc/src/peer_connection/peer_connection_internal.rs

```diff
@@ -1228,27 +1228,25 @@ impl PeerConnectionInternal {
                 Some(r) => r,
                 None => continue,
             };
-            let mid = match transeiver.mid().await {
-                m if !m.is_empty() => m,
-                _ => continue,
-            };
 
-            let tracks = receiver.tracks().await;
+            if let Some(mid) = transeiver.mid() {
+                let tracks = receiver.tracks().await;
 
-            for track in tracks {
-                let track_id = track.id().await;
-                let kind = match track.kind() {
-                    RTPCodecType::Unspecified => continue,
-                    RTPCodecType::Audio => "audio",
-                    RTPCodecType::Video => "video",
-                };
+                for track in tracks {
+                    let track_id = track.id().await;
+                    let kind = match track.kind() {
+                        RTPCodecType::Unspecified => continue,
```

Comment:
Ugh, I really don't like these `Unspecified` enum cases... Need to get rid of those in favour of `Option<Enum>`

Reply:
In my opinion, a separate PR is needed to refuse `Unspecified`.
If we make `kind -> Option<Enum>` it will be out of spec

Reply:
Yeah for sure, I'm just whining :D